_posts/2024-02-18-how-its-made.md

```diff
@@ -0,0 +1,185 @@
+
+![List of commits on GitHub]({% link /static/img/post_2024-02-18-how-its-made/commits.png %}){:width="100%"}
+
+Our GitHub page is also where things are tracked that are not yet part of the game and still in development.
+You might have encountered it before as the place where you can report an issue with the game.
+Besides the git repository of the game itself, GitHub also hosts various other git repository related to the game, like for example OpenGFX, NewGRF development tools or various back-end infrastructure.
+
+## How a feature is born
+
+Now that we know *where* development happens, let's look at *how* it happens.
+
+Each and every change to OpenTTD, no matter if it is a large new feature or just a simple one-line bug fix, starts with someone choosing to spend their time on it.
+As the source code of the game is freely accessible, this can be literally anyone, for a multitude of reasons.
+
+Many features come about because someone has an idea how their game could be better and decides to work on it.
+Bugs are usually fixed because someone either feels responsible for it in some way or is just plain annoyed by it.
+But really, everybody has a different personal motivation.
+There's one reason you're not going to find though: Because the boss said so, or because it is on the big company road map.
+OpenTTD is purely developed by volunteers, there's no company, foundation, or other controlling central entity behind it.
+
+Now let's assume that the person is in fact developing a feature (like for example the new ship pathfinder) and has completed the initial development.
+Since OpenTTD moved to its current home on GitHub almost 6 years ago, we've adopted a fixed process for any change to OpenTTD.
+The first step in the process is to open a so-called pull request (PR) on GitHub with the proposed changes to the source code in the form of git commits.
+Additionally, we require a description of the changes following a specific template to make sure nothing important is left out.
+
+Whenever a PR is opened or updated, a set of automatic checks is kicked off that are designed to catch some easy to make errors.
+The checks can be divided into three groups.
+The first group does some formal checks that are designed to make sure the source code stays readable and consistently formatted.
+The second group triggers a set of various compiles for different compilers and platforms.
+This is to make sure the PR compiles on all platforms we support and not just the one feature was developed on.
+The third group of checks are some automated code security checks that scan the code for common logic errors.
+
+This kind of automated checking are usually referred to as Continuous Integration (CI).
+GitHub will show the result of each check on the PR and also offers a convenient dashboard shown below to dive into all the details.
```

Comment:
This feels like a sidebar that moves the reader's attention away from the big picture of the process. Maybe mention in one sentence that a PR also triggers some automatic checks, but those will be described in detail later.

> When a PR is opened, various automatic checks run to verify some formal things, but while those checks are running, other contributors can already begin to review and comment on the pull request. The automated checks include what is called _continuous integration_ (CI), we will talk a bit more about that further down.

Reply:
Well, the sidebar was mostly just deleted.